### PR TITLE
docs: fix typos discovered by codespell

### PR DESCRIPTION
**Pull request by cclauss**
## Description

% `codespell --ignore-words-list=ans,crate,inout,ratatui,ser,stayin,swarmin,worl`
* https://github.com/codespell-project/codespell

## Change checklist

- [x] Self-review.
- [ ] Documentation updates following the [style guide](https://rust-lang.github.io/rfcs/1574-more-api-documentation-conventions.html#appendix-a-full-conventions-text), if relevant.
- [ ] Tests if relevant.
- [ ] All breaking changes documented.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -848,7 +848,7 @@ All notable changes to iroh will be documented in this file.
 
 - *(*)* Rework `NodeAddr` ([#1506](https://github.com/n0-computer/iroh/issues/1506)) - ([f16e439](https://github.com/n0-computer/iroh/commit/f16e43966e105a2761e18015d5522724bc63ecb6))
 - *(iroh)* Leverage `strum` for `IrohPaths` ([#1507](https://github.com/n0-computer/iroh/issues/1507)) - ([95dce40](https://github.com/n0-computer/iroh/commit/95dce40f7dabd3baeedb07a9e065cbdc7f7226a2))
-- *(iroh,iroh-bytes)* Replace `currrent` with `current` ([#1467](https://github.com/n0-computer/iroh/issues/1467)) - ([4f4d8e5](https://github.com/n0-computer/iroh/commit/4f4d8e5b4366362b9ee5d383ad24355b8cf6a4c5))
+- *(iroh,iroh-bytes)* Replace `current` with `current` ([#1467](https://github.com/n0-computer/iroh/issues/1467)) - ([4f4d8e5](https://github.com/n0-computer/iroh/commit/4f4d8e5b4366362b9ee5d383ad24355b8cf6a4c5))
```
Comment (cclauss) [inline CHANGELOG.md:851]: It is a bad idea to add misspelled words to the CHANGELOG...
```suggestion
- *(iroh,iroh-bytes)* Replace `currrent` with `current` ([#1467](https://github.com/n0-computer/iroh/issues/1467)) - ([4f4d8e5](https://github.com/n0-computer/iroh/commit/4f4d8e5b4366362b9ee5d383ad24355b8cf6a4c5))
```

**Comment by b5** [comment]
thanks for this @cclauss! others, should we add codespell as a CI step? It's helped us on the website repo

File: CHANGELOG.md [hunk 2]
```diff
@@ -22,7 +22,7 @@ All notable changes to iroh will be documented in this file.
 ### 🚜 Refactor
 
 - *(iroh)* Make use of quic-rpc-derive macros to prettify the rpc declarations ([#2508](https://github.com/n0-computer/iroh/issues/2508)) - ([026baaa](https://github.com/n0-computer/iroh/commit/026baaafdf9bd89feef8db4bfedfe9d9853551f3))
-- *(iroh-net)* [**breaking**] Move relay implemention in `iroh-net` behind `iroh-relay` cfg flag ([#2516](https://github.com/n0-computer/iroh/issues/2516)) - ([f37d9f9](https://github.com/n0-computer/iroh/commit/f37d9f9f339dd0de50b35a405ce203c36272a0b4))
```
Comment (dignifiedquire) [inline CHANGELOG.md:25]: please revert the changes in here, 
this file needs to be ignored, as it is regenerated on release based on commit messages

**Review by dignifiedquire — APPROVED** [review]
thanks
